spec/lib/rufo/formatter_source_specs/2.3/squiggly_heredoc.rb.spec

```diff
@@ -144,5 +143,5 @@ EOF
 #~# EXPECTED
 
 <<~EOF
-#{1 }#{2}
```

Comment:
Shouldn't expected here be
```ruby
<<~EOF
#{1} #{2}
EOF
```
?

Reply:
Ah, I think I'm beginning to understand the Ripper bug in 2.3. Please correct me if I'm wrong but it seems from experimentation that when Ripper encounters two interpolation expressions with whitespace between them, it removes the <leading whitespace count> whitespace characters from the space between them, as though the 1st expression is immediately followed by a newline.

It's pretty clear in this example:

``` ruby -rripper -rpp -e 'pp Ripper.lex("<<~EOF\n   \#{1}      \#{2}\nEOF")'
[[[1, 0], :on_heredoc_beg, "<<~EOF"],
 [[1, 6], :on_nl, "\n"],
 [[2, 3], :on_embexpr_beg, "\#{"],
 [[2, 3], :on_tstring_content, ""],
 [[2, 5], :on_int, "1"],
 [[2, 6], :on_embexpr_end, "}"],
 [[2, 10], :on_tstring_content, "   "],
 [[2, 13], :on_embexpr_beg, "\#{"],
 [[2, 15], :on_int, "2"],
 [[2, 16], :on_embexpr_end, "}"],
 [[2, 17], :on_tstring_content, "\n"],
 [[3, 0], :on_heredoc_end, "EOF"]]
```
Note that the heredoc is indented by 3 spaces, and there are 6 spaces between expression 1 and 2, but in Rippers output, there are only 3 spaces between expression 1 and 2.

Reply:
Yes - the sexp doesn't help either:
```[:program,
 [[:string_literal,
   [:string_content,
    [:@tstring_content, "", [2, 3]],
    [:string_embexpr, [[:@int, "1", [2, 5]]]],
    [:@tstring_content, "   ", [2, 10]],
    [:string_embexpr, [[:@int, "2", [2, 15]]]],
    [:@tstring_content, "\n", [2, 17]]]]]]```

Reply:
It turns out fixes have been backported to v3.5, v4. 2 and v5-dev and I now have a fix over those versions (early versions are unfixable). I'll tidy up later.

Reply:
(For others benefit, these refer to Ruby 2.X versions, as in, Ruby 2.3.5, 2.4.2 and 2.5.0-dev. Took me a moment!)

Glad to hear they're being backported to things we use in production. So are you saying on 2.3.5 this test would pass?:

#~# ORIGINAL heredoc_squiggly_extra_spaces
```ruby
<<~EOF
#{1} #{2}
EOF

#~# EXPECTED

<<~EOF
#{1} #{2}
EOF
```

Reply:
@bessey thanks for the version clarification :blush: . Yes all latest versions pass.